File: Xjie/Xjie/Models/SettingsModels.swift
import Foundation

// MARK: - 设置

struct UserSettings: Decodable {
    let intervention_level: String?
    let daily_reminder_limit: Int?
    let glucose_unit: String?
}

struct UpdateSettingsBody: Encodable {
    let intervention_level: String?
    var glucose_unit: String? = nil
}

struct UpdateConsentBody: Encodable {
    let allow_ai_chat: Bool?
    let allow_data_upload: Bool?
}

File: Android/app/src/main/java/com/xjie/app/core/model/SettingsModels.kt
package com.xjie.app.core.model

import kotlinx.serialization.Serializable

@Serializable
data class UserSettings(
    val intervention_level: String? = null,
    val daily_reminder_limit: Int? = null,
    val glucose_unit: String? = null,
)

@Serializable
data class UpdateSettingsBody(
    val intervention_level: String? = null,
    val glucose_unit: String? = null,
)

@Serializable
data class UpdateConsentBody(
    val allow_ai_chat: Boolean? = null,
    val allow_data_upload: Boolean? = null,
)
